crates/gitbutler-command-context/src/repository.rs

```diff
@@ -70,4 +71,51 @@ impl ProjectRepository {
     pub fn repo(&self) -> &git2::Repository {
         &self.git_repository
     }
+
+    /// Fetches a branches name without the remote name attached
+    ///
+    /// refs/heads/my-branch -> my-branch
+    /// refs/remotes/origin/my-branch -> my-branch
```

Comment:
@Byron you can actually have forward slashes in remote names 
![image](https://github.com/user-attachments/assets/576dc8ef-f246-490b-b545-959650a1716a)

Reply:
our `struct Refname` parsing is actually incorrect as it makes that forward slash assumption

Reply:
I see! Then I'd definitely at least mention this in the examples list so it's clear why it is what it is. Of course, a test would be even better, but it's a bit cumbersome to do right now.

Reply:
And I just verified that `gitoxide` sees that similarly, as [the validation](https://github.com/Byron/gitoxide/blob/f8ce3d0721b6a53713a9392f2451874f520bc44c/gix/src/remote/name.rs#L20-L26) (as taken from Git) is quite 'interesting' :D - the name is valid if it can become a valid refspec of the 'normal' form.

Reply:
Git is full of all of these surprises.

For fun the other day ago, I was looking at parsing odb objects for fun. Blobs are just compressed strings, commits are just compressed strings, but trees? Hah! enjoy a bisque format you need to parse 😬 https://gist.github.com/Caleb-T-Owens/e8630f2b600e99acc3c3e317efb42a61.

I spent about 2 hours reverse engineering that format 😆.

Some day I'll try to understand packfiles...

Reply:
> Some day I'll try to understand packfiles...

They are a bit more complicated than trees especially if you plan to decode objects as well. And if reading the Git sources isn't cheating, that would definitely be my recommendation (but where is the fun in that ;)).

Reply:
I've found a well written post that goes over how to read packfiles in haskell so I'll probably make good use of that https://stefan.saasen.me/articles/git-clone-in-haskell-from-the-bottom-up/#pack-file-format